File: backend/app/schemas/provider.py
"""Provider connection API schemas."""

from uuid import UUID

from pydantic import BaseModel

from app.models.repository import RepositoryPlatform


class RepositoryProviderStatus(BaseModel):
    """Provider readiness for publishing fixes from one tracked repository."""

    repository_id: UUID
    provider: RepositoryPlatform | None
    is_connected: bool
    can_publish: bool
    installation_id: str | None = None
    account_login: str | None = None
    message: str

File: frontend/src/types/provider.ts
import type { RepositoryPlatform } from "@/types/repository";

export type RepositoryProviderStatus = {
  repository_id: string;
  provider: RepositoryPlatform | null;
  is_connected: boolean;
  can_publish: boolean;
  installation_id?: string | null;
  account_login: string | null;
  message: string;
};
